### PR TITLE
libobs: Fix Monitoring devices showing input devices

### PR DESCRIPTION
**Pull request by TheOrsa**
The main bug fix is to show output devices in the Advanced Audio
Settings configuration, where currently on macOS it shows erroneously
input devices.

Second and third bugfixes are minor: after retrieving audio device UID
we then check for errors and return if we were not able to retrieve
the data. However cf_uid may already be allocated (since
AudioObjectGetPropertyData is OSX Api it may allocate then return
error. Checking for it being non-null and freeing if it is doesn't
hurt.

Last fix is the lowercase file not found error on coreaudio-output.c
file. Default MacOSX partitioning is case-insensitive, thus
by most developers. However, the actual dir name is AudioToolbox.
In other words, this bug is only exposed for developers compiling
on HFS case-sensitive based-storage.

#### TESTING #######

Main bug fix tested by using 2 output devices, one (now able to) selected as monitor
When selecting "Monitor and Output" or "Monitor only" it now routes audio to selected
device.
Manually able to compile OBS Project on case-sensitive file system, proving last fix
also works.

### CODE REVIEW

**Comment by pkviet** [comment]
i've wanted to investigate this bug for a long time; good work.
IMO you should separate in 3 different commits since there are three bugfixes.
When you're done just force push your local branch to your fixMonitoringOnOSX ; this will update the PR automatically (no need to close it)

**Comment by TheOrsa** [comment]
Thanks for reviewing, pkviet! I have splitted into 3 commits so fixes are more clear and separate.
Please let me know any other suggestion!

**Comment by jp9000** [comment]
Please change commit titles to:

`libobs: Fix macOS monitoring devices showing input devices`
`libobs: Add macOS check for cf_uid pointer free`
`libobs: Fix compilation issue on case-sensitive filesystems`

**Comment by TheOrsa** [comment]
Thank you for that!

**Comment by RytoEX** [comment]
Super minor nitpicks:

1.  There's an extra space in the first commit's message body:
```
This  bug
```
2.  The second commit message body is missing a period at the end of it.

**Comment by TheOrsa** [comment]
Thanks!
